pkg/frontend/querymiddleware/astmapper/parallel.go

```diff
@@ -28,6 +28,8 @@ var NonParallelFuncs = []string{
 	"absent",
 	"absent_over_time",
 	"histogram_quantile",
+	"limitk",
```

Comment:
I _think_ `limitk` is safe to parallelize, just like topk/bottomk.

Maybe it's worth creating a new `good-first-issue` issue and let someone send a PR adding some tests for it?

Reply:
Topk and bottok are not parallelizeable as well ([see details in this internal ticket](https://github.com/grafana/mimir-squad/issues/270#issuecomment-918920983)).

Why they're not mentioned here? Right now I don't remember all implementation details of how these functions are guaranteed not to be parallelized. I will investigate it further, but will not block this PR on this.